src/service/handlers/prepared.rs

```diff
@@ -276,21 +276,78 @@ pub(crate) async fn do_put_prepared_statement_update(
+        // Let the session layer determine the best execution strategy
+        let session_clone = Arc::clone(&session);
+        let sql_clone = sql.clone();
+        let batches_clone = batches.clone();
+        
+        let affected_rows = tokio::task::spawn_blocking(move || {
+            session_clone
+                .execute_with_record_batches(&sql_clone, batches_clone)
+                .map(|rows| rows as i64)
+        })
+        .await
+        .map_err(SwanFlightSqlService::status_from_join)?;
+        
+        match affected_rows {
+            Ok(rows) => {
+                info!(
+                    handle = %handle,
+                    affected_rows = rows,
+                    "statement executed with RecordBatch optimization"
+                );
+                return Ok(rows);
+            }
+            Err(err) => {
+                // If RecordBatch execution failed (e.g., not an INSERT), convert to parameters
+                debug!(%err, "RecordBatch execution not available, converting to parameters");
+                
+                // Convert RecordBatches to parameter sets for fallback
+                let parameter_sets: Vec<Vec<duckdb::types::Value>> = batches
+                    .iter()
+                    .flat_map(|batch| {
+                        SwanFlightSqlService::record_batch_to_params(batch)
+                            .unwrap_or_else(|_| Vec::new())
+                    })
+                    .collect();
+                
+                if parameter_sets.is_empty() {
+                    return Err(SwanFlightSqlService::status_from_error(err));
+                }
+
+                let session_clone = Arc::clone(&session);
+                let affected_rows = tokio::task::spawn_blocking(move || {
+                    SwanFlightSqlService::execute_statement_batches(&sql, &parameter_sets, &session_clone)
+                })
+                .await
+                .map_err(SwanFlightSqlService::status_from_join)?
+                .map_err(SwanFlightSqlService::status_from_error)?;
 
-    let parameter_set_count = parameter_sets.len();
+                info!(
+                    handle = %handle,
+                    affected_rows,
+                    "prepared statement update complete (parameter fallback)"
+                );
+                return Ok(affected_rows);
+            }
+        }
+    }
 
+    // No batches provided - execute with empty parameters
     info!(
         handle = %handle,
         sql = %sql,
-        parameter_sets = parameter_set_count,
-        "executing prepared statement update"
+        "executing prepared statement update without parameters"
     );
 
     let session_clone = Arc::clone(&session);
 
     let affected_rows = tokio::task::spawn_blocking(move || {
-        SwanFlightSqlService::execute_statement_batches(&sql, &parameter_sets, &session_clone)
+        SwanFlightSqlService::execute_statement_batches(&sql, &[], &session_clone)
```

Comment:
so we can remove the parameter_sets parameter from execute_statement_batches method?